### PR TITLE
Biorepo dataloader and training

### PR DESCRIPTION
**Pull request by nathan-cain15**
**Biorepo.py:** code is very similar to hawaii.py

**Dataloaders:** I added scientific_name to the same the dataloaders in order to calculate seen vs unseen species metrics. Using taxonId would probably be better but beetlepalooza doesn’t have it as of now.

**hawaii.py** Fixed logging error when creating informed train/val split.

**train.py**

- Changed plot_preds to take in sample_idx and plot the sample at that index in the batch.
- Added 3 config parameters: n_val_fixed, n_val_worst, and n_val_random to determine what predictions to plot for a validation step
- in validate()
    - finds the top_k worst images to plot
    - added calculating the seen vs unseen metrics (loss, line_err_cm, point_err_cm, max_line_err_cm
- added get_training_species() which compiles a set of all the species in the training data
- in train(): split the validation dataset so validation is done separately on hawaii and biorepo in order to get metrics specific to each dataset. Definitely can refactor if needed, there are twice the amount of metrics now which is starting to seem like a lot.

### CODE REVIEW

File: src/btx/data/biorepo.py [hunk 1]
```diff
@@ -1,32 +1,231 @@
...
+            scalebar = [0.0, 0.0, 1.0, 1.0]
+            
+
+        loss_mask = np.array([1.0, 1.0])  # Train on both width and length
+        msg = f"Expected loss_mask shape (2,), got {loss_mask.shape}"
+        assert loss_mask.shape == (2,), msg
+        return utils.Sample(
+            img_fpath=str(fpath),
+            points_px=np.array(elytra_width + elytra_length).reshape(2, 2, 2),
+            scalebar_px=np.array(scalebar),
```
[the remark this answers is not in the document]
Comment (nathan-cain15) [reply src/btx/data/biorepo.py]: the scalebar is already in the right shape per the annotations
